framework/midi/midievent.h

```diff
@@ -84,13 +84,13 @@ struct Event {
         setOpcode(opcode);
     }
 
-    [[deprecated]] Event(channel_t ch, EventType type)
+    /*[[deprecated]]*/ Event(channel_t ch, EventType type)
```

Comment:
This constructor will be deleted in MU4. That's why it's marked as deprecated and must not be used.

Reply:
It currently is though, and as such leads to compiler warnings.
Marking this as a ToDo is fine, causing compiler warnings is not

Reply:
Furthermore it seems the build on macOS doesn't like that `[[deprecated]]` syntax at all and gets thrown off track by them quite miserably, causing the development builds not to succeed?

Reply:
And indeed `[[deprecated]]` is a C++14 feature (see https://en.cppreference.com/w/cpp/language/attributes/deprecated), but we're not using that (yet), esp. on Mac (but also Linux and MINGW) we're using `-std=c++11`

Reply:
Maybe using `QT_DEPRECATED`, `QT_DEPRECATED_X(sometext)`, `QT_DEPRECATED_VARIABLE` or `QT_DEPRECATED_CONSTRUCTOR` for this instead? At least then it should be working for any compiler?

Reply:
> so many warnings

You can disable deprecation warning in your build settings. Now we are working on updating the code with future remove these deprecated methods. 

> build on macOS

I'm working on macOS, everything is fine

> C++14 feature 

We are moving to C++17 soon. C++11 just ignores it as unknown attribute

> QT_DEPRECATED

This class does not use Qt at all.

Reply:
I know I could disable warnings, but would miss new ones (like with Qt 5.15) that just got introduced recently.
I know that this class doesn't use Qt, but it could use that macro. I tried, it works (but of course still emits a warning)

Reply:
I already fixed most of those warnings, but I have some troubles with building tests. Later I will return to my PR and do something with tests
https://github.com/musescore/MuseScore/pull/6580

Reply:
Reverted to `[[deprecated]]`